go.mod

```diff
@@ -1,15 +1,15 @@
 module github.com/hashicorp/go-msgpack/v2
 
-go 1.19
+go 1.20
```

Comment:
nit: are there any specific benefits we needed from bumping the minimum Go version? I would tend slightly towards leaving it for libraries unless we need to bump it - just to make it as easy as possible for dependents to upgrade, but I'm also not hugely opposed to bumping it.